frontend/viewer/src/project/browse/EntriesList.svelte

```diff
@@ -38,34 +38,49 @@
   projectEventBus.onEntryDeleted(entryId => {
     if (selectedEntryId === entryId) onSelectEntry(undefined);
     if (entriesResource.loading || !entries.some(e => e.id === entryId)) return;
-    entriesResource.mutate(entriesResource.current?.filter(e => e.id !== entryId) ?? []);
+    const currentIndex = entriesResource.current?.findIndex(e => e.id === entryId) ?? -1;
+    if (currentIndex >= 0) {
+      entriesResource.current!.splice(currentIndex, 1);
+    }
   });
   projectEventBus.onEntryUpdated(_entry => {
     if (entriesResource.loading) return;
-    entriesResource.mutate(entriesResource.current?.map(e => e.id === _entry.id ? _entry : e) ?? []);
+    const currentIndex = entriesResource.current?.findIndex(e => e.id === _entry.id) ?? -1;
+    if (currentIndex >= 0) {
+      entriesResource.current![currentIndex] = _entry;
+    } else {
+      void silentlyRefreshEntries();
+    }
   });
 
+  async function silentlyRefreshEntries() {
+    const updatedEntries = await fetchCurrentEntries();
+    entriesResource.mutate(updatedEntries);
```

Comment:
interesting, I think I understand what's going on here. One issue is that this is no longer debounced (or deduplicated), meaning if 10 entry changes come in then it's going to run 10 queries at once. I'm guessing you're doing this so we don't show a loading state, maybe we should control this from that end.